### DISCUSSION TITLE
One line less please! :)

### DISCUSSION THREAD

**Discussion by mcctuxic** [tabemann/zeptoforth#159] [Ideas]
Hi,

when using more-words the listing stops, when one page is filled:
The newest defined word is placed at the first line of the page.
When pressing 'q', because most of the time (I think) one is interested
in the latest stuff done so far, just this "most important line" ;) with
the newest defined word is pushed off screen by printing 'ok'.

Would it be useful to shorten the listing of words by more-words just
by one line in a way, that the the 'q' following 'ok' can be printed without
moving the listing a further line up?

**Comment by tabemann** [comment]
On Tue, Nov 26, 2024 at 10:44 PM mcctuxic ***@***.***> wrote:

> Hi,
>
> when using more-words the listing stops, when one page is filled:
> The newest defined word is placed at the first line of the page.
> When pressing 'q', because most of the time (I think) one is interested
> in the latest stuff done so far, just this "most important line" ;) with
> the newest defined word is pushed off screen by printing 'ok'.
>
> Would it be useful to shorten the listing of words by more-words just
> by one line in a way, that the the 'q' following 'ok' can be printed
> without
> moving the listing a further line up?
>
> I just tested it and see what you mean. The only problem, though, is that
the:

\*** Press q to exit, any other key to continue \***

line would no longer be at the bottom of the screen if one executed PAGE
MORE-WORDS, because it would output less than a screenful of output at a
time.

I thought if it could be possible to suppress the 'ok' generated
afterwards, but as that is not generated by MORE-WORDS itself but rather by
the outer interpreter, the only way of turning it off in this scenario is
to raise an exception or otherwise call QUIT or ABORT, but those would have
undesirable consequences (e.g. if MORE-WORDS was called within another word
it would cause that word to be exited out of without being completed, and
ABORT or an exception would cause a BEL to be transmitted).

Travis
